### PR TITLE
Miscellaneous corrections, some POSIX improvements

### PR DESCRIPTION
**Pull request by Juesto**
As you can see on the commits and discussed/experienced.
* just one leftover from ps3 so far on the vpc support (a PS3 port never happened on Valve side)
* removed some of the unnecessary stuff from vpc projects and groups.
* protobuf path corrected on posix
* sys_linux.cpp was duplicated for some reason and excluded on all platforms, meaning it got affected by posix. 
* correctly guarded sal.h for posix so it doesn't cause null related issues over there. (sorry maxim, at least it's cleaner)
* needed executable bit for binary files used in POSIX
* macos should prefer base sdk instead of something specific.

Is it okay to make combo PRs?

### CODE REVIEW

**Comment by maximsmol** [comment]
What is the point of the VPC changes in the first place?

We also specifically want to avoid removing support for PS3/XBOX and other obscure features.

**Comment by micwoj92** [comment]
This branch has conflicts that must be resolved

**Comment by Juesto** [comment]
> This branch has conflicts that must be resolved

They're better than what's committed. Done

**Comment by Juesto** [comment]
Added protobuf path fix, noticed minor additional issues.

**Comment by Juesto** [comment]
> We also specifically want to avoid removing support for PS3/XBOX and other obscure features.

You can always add it back later on when it actually gets worked on.

PS3 is currently not workable due to lots of missing files unfortunately. There never was a PS3 version from Valve to begin with.



> What is the point of the VPC changes in the first place?

optimization by cleaning up non-existing projects and adjusting groups accordingly. How do you miss what I said?

**Comment by Juesto** [comment]
Updated: "rebased"/merged from current master pretty much, no significant changes here.

**Comment by maximsmol** [comment]
This PR changes random things that are either already in `build-linux` or just completely unused anyway. We deliberately avoid making cleanup changes for their own sake because it introduces a lot of noise.

**Comment by Juesto** [comment]
it doesn't introduce much noise, could revert projects if you want, it does enhance groups for expanding ability to build more projects.
yes i am making official the good changes of build-linux that also benefit mac
please do request changes, i am open for cooperation.

**Comment by melvyn2** [comment]
Targeting the base sdk (for macos) is a *very* bad idea, as TF2 relies on old code and 32 bit support.

**Comment by Juesto** [comment]
> Targeting the base sdk (for macos) is a _very_ bad idea, as TF2 relies on old code and 32 bit support.

Xcode does not include multiple SDKs (particularly very old sdks) since the App Store release. any requirement for specific projects can be done within the respective project vpc file.
the default sdk works fine in most cases, it is the recommended usage. not forcing the usage of very specific sdks where the availability is dubious.

**Comment by melvyn2** [comment]
These sdks are required for the build, so someone building this would have them anyways, or would still need to. Additionally, if you add this change in this PR, then it's up to you to add those sdk's where they're needed.

**Comment by Juesto** [comment]
> These sdks are required for the build, so someone building this would have them anyways, or would still need to. Additionally, if you add this change in this PR, then it's up to you to add those sdk's where they're needed.

are you sure you know what are you talking about?
the very specific sdk is not required, its supposed to compile fine with the default sdk the xcode version ships with.

**Comment by melvyn2** [comment]
> > These sdks are required for the build, so someone building this would have them anyways, or would still need to. Additionally, if you add this change in this PR, then it's up to you to add those sdk's where they're needed.
> 
> are you sure you know what are you talking about?
> the very specific sdk is not required, its supposed to compile fine with the default sdk the xcode version ships with.

No it's not? The latest sdk's don't even include 32 bit libraries, so there's no way tf2 can compile with them.

**Comment by Juesto** [comment]
> No it's not? The latest sdk's don't even include 32 bit libraries, so there's no way tf2 can compile with them.

You don't compile in the latest Xcode version until 64 bits is implemented.

**Comment by melvyn2** [comment]
> > No it's not? The latest sdk's don't even include 32 bit libraries, so there's no way tf2 can compile with them.
> 
> You don't compile in the latest Xcode version until 64 bits is implemented.

Still, relying on an unversioned sdk can cause all sorts of issues, so I'd rather see a versioned sdk used. Additionally, `vgui` does require `macosx10.5` sdk.

**Comment by Juesto** [comment]
> Still, relying on an unversioned sdk can cause all sorts of issues, so I'd rather see a versioned sdk used. Additionally, `vgui` does require `macosx10.5` sdk.

macos is broken for the time being, vgui_surfacelib specifically is the one that needs 10.5 sdk, this should be done on the vpc file, not hardcoding sdks for all projects. but we dont know if it even has or accepts a value for specifying osx sdks.
for the time being you only build fine with xcode 9 since it enables c++17 and is 32 bits.

**Comment by Juesto** [comment]
Is it possible for PRs to be cherry-picked or selectively exclude commits?

**Comment by micwoj92** [comment]
Just reverse commits

File: creategameprojects.bat [hunk 1]
```diff
@@ -1,2 +1,3 @@
-devtools\bin\vpc.exe /2015 /define:WORKSHOP_IMPORT_DISABLE /define:SIXENSE_DISABLE /define:NO_X360_XDK /define:RAD_TELEMETRY_DISABLED /define:DISABLE_ETW /define:LTCG /no_ceg /retail /define:CERT /tf +game /mksln games.sln
-@PAUSE
+@set VALVE_NO_AUTO_P4=1
```
Comment (Margen67) [inline creategameprojects.bat:1]: Isn't this redundant with `/nop4add`?

**Comment by micwoj92** [comment]
This branch has conflicts that must be resolved!

**Comment by Juesto** [comment]
> This branch has conflicts that must be resolved!

stop reminding me @micwoj92

**Pull request closed without merging**